src/main/java/org/springframework/integration/dsl/http/HttpMessageHandlerSpec.java

```diff
@@ -182,11 +182,55 @@ public HttpMessageHandlerSpec uriVariable(String variable, Expression value) {
 		return this;
 	}
 
+	/**
+	 * @param variable the uri template variable.
+	 * @param value the expression to evaluate value for te uri template variable.
+	 * @return the current Spec.
+	 * @see HttpRequestExecutingMessageHandler#setUriVariableExpressions(Map)
+	 * @since 1.1.1
+	 */
+	public HttpMessageHandlerSpec uriVariable(String variable, String value) {
+		return uriVariable(variable, PARSER.parseExpression(value));
+	}
+
+	/**
+	 * @param variable the uri template variable.
+	 * @param valueFunction the function to evaluate value for te uri template variable.
+	 * @param <P> the payload type.
+	 * @return the current Spec.
+	 * @see HttpRequestExecutingMessageHandler#setUriVariableExpressions(Map)
+	 * @since 1.1.1
+	 */
+	public <P> HttpMessageHandlerSpec uriVariable(String variable, Function<Message<P>, ?> valueFunction) {
+		return uriVariable(variable, new FunctionExpression<Message<P>>(valueFunction));
+	}
+
 	public HttpMessageHandlerSpec uriVariablesExpression(Expression uriVariablesExpression) {
 		this.messageHandler.setUriVariablesExpression(uriVariablesExpression);
 		return this;
 	}
 
+	/**
+	 * @param uriVariablesExpression to use.
+	 * @return the current Spec.
+	 * @see HttpRequestExecutingMessageHandler#setUriVariablesExpression(Expression)
+	 * @since 1.1.1
+	 */
+	public HttpMessageHandlerSpec uriVariablesExpression(String uriVariablesExpression) {
```

Comment:
Plus here.
I dug the whole project and haven't found so much much vulnerabilities around `string expression representations`.